kernel/poll.c

```diff
@@ -424,7 +424,7 @@ static int signal_poll_event(struct k_poll_event *event, uint32_t state)
 		} else if (poller->mode == MODE_TRIGGERED) {
 			retcode = signal_triggered_work(event, state);
 		} else {
-			;
+			; /* Non-empty final else in if ... else if struct */
```

Comment:
And here, does is this ok? Can we do better?
`Poller is not set as poll or triggered mode, so no action needed.`

Reply:
Similarly:

```c
if (poller->mode == MODE_POLL) {
    retcode = signal_poller(event, state);
}
if (poller->mode == MODE_TRIGGERED) {
    retcode = signal_triggered_work(event, state);
}
```

If mode is TRIGGERED, it is clearly not also POLL (unless the two constants are equal, which they aren't).  To the compiler, the else was logically superfluous to begin with, better to remove it than add yet more superfluity.

Reply:
same comment as above. I've been instructed to revise the comments not refactor, as that is for code owners.